public/app/features/explore/Logs/Logs.tsx

```diff
@@ -156,81 +140,88 @@ const getDefaultVisualisationType = (): LogsVisualisationType => {
-  cancelFlippingTimer?: number;
-  topLogsRef = createRef<HTMLDivElement>();
-  logsVolumeEventBus: EventBus;
-  static contextType = ContentOutlineContext;
-  declare context: React.ContextType<typeof ContentOutlineContext>;
+const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
+  const {
+    width,
+    splitOpen,
+    logRows,
+    logsMeta,
+    logsVolumeEnabled,
+    logsVolumeData,
+    loadLogsVolumeData,
+    loading = false,
+    onClickFilterLabel,
+    onClickFilterOutLabel,
+    timeZone,
+    scanning,
+    scanRange,
+    showContextToggle,
+    absoluteRange,
+    onChangeTime,
+    getFieldLinks,
+    theme,
+    logsQueries,
+    clearCache,
+    addResultsToCache,
+    exploreId,
+    getRowContext,
+    getLogRowContextUi,
+    getRowContextQuery,
+    loadMoreLogs,
+    panelState,
+    eventBus,
+  } = props;
+  const [showLabels, setShowLabels] = useState<boolean>(store.getBool(SETTINGS_KEYS.showLabels, false));
+  const [showTime, setShowTime] = useState<boolean>(store.getBool(SETTINGS_KEYS.showTime, true));
+  const [wrapLogMessage, setWrapLogMessage] = useState<boolean>(store.getBool(SETTINGS_KEYS.wrapLogMessage, true));
+  const [prettifyLogMessage, setPrettifyLogMessage] = useState<boolean>(
+    store.getBool(SETTINGS_KEYS.prettifyLogMessage, false)
+  );
+  const [dedupStrategy, setDedupStrategy] = useState<LogsDedupStrategy>(LogsDedupStrategy.none);
+  const [hiddenLogLevels, setHiddenLogLevels] = useState<LogLevel[]>([]);
+  const [logsSortOrder, setLogsSortOrder] = useState<LogsSortOrder>(
+    store.get(SETTINGS_KEYS.logsSortOrder) || LogsSortOrder.Descending
+  );
+  const [isFlipping, setIsFlipping] = useState<boolean>(false);
+  const [displayedFields, setDisplayedFields] = useState<string[]>([]);
+  const [forceEscape, setForceEscape] = useState<boolean>(false);
+  const [contextOpen, setContextOpen] = useState<boolean>(false);
+  const [contextRow, setContextRow] = useState<LogRowModel | undefined>(undefined);
+  const [visualisationType, setVisualisationType] = useState<LogsVisualisationType | undefined>(
+    panelState?.logs?.visualisationType ?? getDefaultVisualisationType()
+  );
+  const [logsContainer, setLogsContainer] = useState<HTMLDivElement | undefined>(undefined);
+  const dispatch = useDispatch();
+  const previousLoading = usePrevious(loading);
+
+  const logsVolumeEventBus = eventBus.newScopedBus('logsvolume', { onlyLocal: false });
+  const { outlineItems, register, unregisterAllChildren } = useContentOutlineContext() ?? {};
+  let flipOrderTimer: number | undefined = undefined;
+  let cancelFlippingTimer: number | undefined = undefined;
   // @ts-ignore
-  private toggleLegendRef: React.MutableRefObject<(name: string, mode: SeriesVisibilityChangeMode) => void> =
+  const toggleLegendRef: React.MutableRefObject<(name: string, mode: SeriesVisibilityChangeMode) => void> =
     React.createRef();
+  const topLogsRef = React.createRef<HTMLDivElement>();
```

Comment:
I think these could be `useRef()`.

Reply:
Fixed in 06dcdc8ca9d7344131a2f3e0ed081f463bde4a08